### DISCUSSION TITLE
Enable to determine file edit

### DISCUSSION THREAD

**Discussion by MagaKhachatryan** [dokan-dev/dokan-dotnet#321] [Q&A] [answered]
Dear Dokan support team,
we know that Dokan gives us the possibility to be notified when the file is moved, copied, deleted, written in, etc.
On the other hand, we can catch the point of writing, and reading in a file.

But in some cases, the whole of these sets of events is not enough to determine file edit.
For example :
1. we are opening .csv file via Excel
2. adding some text
3. saving it

It looks like Excel is not throwing the SetEndOfFile event ( like .txt editors ) at the end of the whole editing operation and we are not able to determine where editing is finished to start processing the edited file.

Can you provide some hints on how to determine that editing is finished (not only for the Microsoft editor's case but something very general for all types of files/ editors)?

Thanks in advance.

**Comment by Liryna** [comment, marked as answer]
Hi @MagaKhachatryan ,

As all the activity (Create / Read / Write / SetEndOfFile) directly comes from the application manipulating the file and they are all written by different developers, there is not generic workflow on how files are edited.
Maybe a better logic would be to observe the file activity and determine that after a delay of X ms / sec of no handle with previous writes activity opened, the editor is done.

FYI Word (Office) use a tmp and source file replacement workflow when saving. If you catch this pattern, you can say when Office is done to apply the changes. I also believe it keeps a tmp file opened side to side to the original file while the user edits (need to confirm).

**Comment by MagaKhachatryan** [comment]
Thank you for your answer @Liryna  very much!!!

Now we do so for Word files but we have found out it doesn't work for .csv files.
We have decided to apply a more general solution.
Our team has thought about the solution you suggested but we wanted to make sure that there is no other way. 
So we'll apply this solution.

Thank you very much to the whole Dokan team. 
It is a great pleasure to work with you.
